src/d3a/models/myco_matcher/pay_as_bid.py

```diff
@@ -0,0 +1,37 @@
+from d3a.constants import FLOATING_POINT_TOLERANCE
+from d3a.models.market.market_structures import BidOfferMatch
+from d3a.models.myco_matcher.base_matcher import BaseMatcher
+
+
+class PayAsBidMatch(BaseMatcher):
+
+    def calculate_match_recommendation(self, bids, offers, current_time=None):
+        """
+        Pay as bid first
+        There are 2 simplistic approaches to the problem
+        1. Match the cheapest offer with the most expensive bid. This will favor the sellers
+        2. Match the cheapest offer with the cheapest bid. This will favor the buyers,
+           since the most affordable offers will be allocated for the most aggressive buyers.
+        """
+
+        return self._perform_pay_as_bid_match(bids, offers)
+
+    def _perform_pay_as_bid_match(self, bids, offers):
+        # Sorted bids in descending order
+        sorted_bids = self.sort_by_energy_rate(bids, True)
+        # Sorted offers in descending order
+        sorted_offers = self.sort_by_energy_rate(offers, True)
+        already_selected_bids = set()
+        bid_offer_pairs = []
+        for offer in sorted_offers:
+            for bid in sorted_bids:
+                if bid.id in already_selected_bids or offer.seller == bid.buyer:
+                    continue
+                if (offer.energy_rate - bid.energy_rate) <= FLOATING_POINT_TOLERANCE:
+                    already_selected_bids.add(bid.id)
+                    selected_energy = min(bid.energy, offer.energy)
+                    bid_offer_pairs.append(BidOfferMatch(
+                        bid=bid, offer=offer, bid_energy=selected_energy,
+                        offer_energy=selected_energy, trade_rate=bid.energy_rate))
```

Comment:
In both pay_as_bid and pay_as_clear we passed bid_energy and offer_energy where they were equal 
If this is the case we can use 1 variable instead of two

Reply:
`BidOfferMatch` was there before this PR, though I agree that both `bid_energy/offer_energy` could be merged to `selected_energy`